main.tf

```
@@ -0,0 +1,11 @@
+resource "aws_ebs_encryption_by_default" "this" {
+  enabled = var.ec2_account_settings.aws_ebs_encryption_by_default.enabled
+}
```

Need to also manage the resource, `aws_ebs_default_kms_key`, so we can set the kms key to use.

```suggestion
resource "aws_ebs_encryption_by_default" "this" {
  enabled = var.ec2_account.ebs_encryption_by_default.enabled
}

resource "aws_ebs_default_kms_key" "this" {
  count = var.ec2_account.ebs_default_kms_key_arn != null ? 1 : 0

  key_arn = var.ec2_account.ebs_default_kms_key_arn
}
```